### ISSUE TITLE
merge branches, job running and locks

### ISSUE THREAD

**Issue by martinmart92** [b2ihealthcare/snow-owl#484]
Good Morning,
I am getting the following issue with branch merging in SnowOwl 7.x.
Starting from an empty index, I imported Snomed-International in the `MAIN` branch.
I then imported Snomed-UK (snapshot) into `MAIN/2018-07-31` (a sub-branch of Snomed-International).
I then created an "empty" branch `A` starting from `MAIN/2018-07-31`: the new branch path is `MAIN/2018-07-31/A`

Now since I want to use branch `A` to have visibility and create concepts that may refer to both Snomed-UK and Snomed-International concepts, I tried to merge Snomed-UK branch into branch A. 
(Would that be a reasonable expectation? Or possible scenario?)

After the merge, I still have the job in RUNNING status if I query GET `/snowowl/admin/jobs`, and Elasticsearch index "locks-lock" contains two  `synchronizing changes` (one for source and destination branches involved in the merge). This prevents me for further merge into the branch A:  I wish to also import and merge Snomed-US snapshot and merge it into A, but a `GET /snowowl/snomed-ct/v3/merges/{id}` returns a status failed with "Could not acquire requested lock(s)."

Is this a bug? How to make SnowOwl "finish" the merge and have the destination branch unlocked for subsequent merges?

Thanks!

**Comment by cmark** [comment]
Hi @martinmart92,

The branch `MAIN/2018-07-31` was created by you or by the RF2 importer? 
If by the RF2 importer, you could create a new `SNOMEDCT-UK` child branch under the `2018-07-31` branch to separate the version branch and the UK content properly. 
Then you could create the branch `A` under the SNOMEDCT-UK child branch, so at the end it would look like this `MAIN/2018-07-31/SNOMEDCT-UK/A`.
At this point you don't need to merge anything into the `A` branch since you have created it just after importing the UK content into its parent branch `MAIN/2018-07-31/SNOMEDCT-UK`.
This is what we do when it comes to maintaining multiple extensions in a single Snow Owl dataset.

I'll take a look at why do you have such a long running merge operation that potentially does nothing at all, or is already completed but did not release the locks. According to our huge number of merge and rebase test cases this should not happen at all, but there could be something that I'm missing here that would indicate a missing scenario in our test suite.

Will get back to you once I have something to share.

Cheers,
Mark

**Comment by cmark** [comment]
Hi @martinmart92,

One questions, did a server restart happen before the merge could finish (either accidentally or intentionally)?
If yes, then that is a bug in the server, after restart there might be stale locks lying around and those need to be cleaned during boot.
If you encounter such locks, you can force unlock them using the CLI available after you `telnet` or `ssh` into the OSGi console.
Example: 

```
$ telnet localhost 2501
$ osgi> snowowl locks
$ osgi> snowowl locks remove -f <lockIdOrTarget>
```

This is planned to be fixed in the upcoming 7.6.0 release.

Cheers,
Mark
